### DISCUSSION TITLE
dynamic routes without nesting /

### DISCUSSION THREAD

**Discussion by MaksZhukov** [vercel/next.js#43133] [Help] [answered]
Hey! Is there a functionality to have a dynamic route like this: 
in pages folder i want to have a file product-[type]-[slug] and if i open a page in browser /product-wheel-torcue-1 I want to see the dynamic page without nesting "/"

**Comment by icyJoseph** [comment, marked as answer]
I think, if you use `getServerSideProps`, you could use rewrites:

```js
/** @type {import('next').NextConfig} */
const nextConfig = {
  reactStrictMode: true,
  async rewrites() {
    return [
      {
        source: "/product-:type(\\w+)-:slug(\\w+)",
        destination: "/", // can be anywhere else :) maybe /product
      },
    ];
  },
};

module.exports = nextConfig;

```

And then in `pages/index.js`, when you go to `http://localhost:3000/product-foo-bar`

```tsx
import { GetServerSidePropsContext } from "next";
import { useRouter } from "next/router";

export default function Home() {
  console.log(useRouter().query); // { type: 'foo', slug: 'bar' }
  return <div>Product</div>;
}

export const getServerSideProps = (ctx: GetServerSidePropsContext) => {
  console.log(ctx.query); // { type: 'foo', slug: 'bar' }
  return { props: {} };
};

```

You could run a simple validation in `getServerSideProps` to make sure type and slug are present and valid, and if not return `{notFound: true}`.
